include/zephyr/drivers/bluetooth.h

```diff
@@ -0,0 +1,253 @@
+};
+
+/**
+ * @brief Open the HCI transport.
+ *
+ * Opens the HCI transport for operation. This function must not
+ * return until the transport is ready for operation, meaning it
+ * is safe to start calling the send() handler.
+ *
+ * @param dev  HCI device
+ * @param recv Callback to receive data from the controller
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_open(const struct device *dev, bt_hci_recv_t recv)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	return api->open(dev, recv);
+}
+
+/**
+ * @brief Close the HCI transport.
+ *
+ * Closes the HCI transport. This function must not return until the
+ * transport is closed.
+ *
+ * @param dev HCI device
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_close(const struct device *dev)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	if (api->close == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->close(dev);
+}
+
+/**
+ * @brief Send HCI buffer to controller.
+ *
+ * Send an HCI command or ACL data to the controller. The exact
+ * type of the data can be checked with the help of bt_buf_get_type().
+ *
+ * @note This function must only be called from a cooperative thread.
+ *
+ * @param dev HCI device
+ * @param buf Buffer containing data to be sent to the controller.
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_send(const struct device *dev, struct net_buf *buf)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	return api->send(dev, buf);
+}
+
+#if defined(CONFIG_BT_HCI_SETUP) || defined(__DOXYGEN__)
+/**
+ * @brief HCI vendor-specific setup
+ *
+ * Executes vendor-specific commands sequence to initialize
+ * BT Controller before BT Host executes Reset sequence.
+ *
+ * @note @kconfig{CONFIG_BT_HCI_SETUP} must be selected for this
+ * field to be available.
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_setup(const struct device *dev, struct bt_hci_setup_params *params)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	if (api->setup == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->setup(dev, params);
+}
+#endif
+
+/**
+ * @}
+ */
+
+/* The following functions are not strictly part of the HCI driver API, in that
+ * they do not take as input a struct device which implements the HCI driver API.
+ */
+
+/**
+ * @brief Setup the HCI transport, which usually means to reset the
+ * Bluetooth IC.
+ *
+ * @note A weak version of this function is included in the H4 driver, so
+ *       defining it is optional per board.
+ *
+ * @param dev The device structure for the bus connecting to the IC
+ *
+ * @return 0 on success, negative error value on failure
+ */
+int bt_hci_transport_setup(const struct device *dev);
```

Comment:
Is there a reason for leaving the transport setup and teardown out of the driver API? It looks like these will always used together with `open` and `close`. Maybe we should utilize the setup API to pass in the HCI transport device to be used in `open` and `close`?

Reply:
This is really just something inherited from the original `include/drivers/bluetooth/hci_driver.h`. This API should probably be moved somewhere else since it's really an extension that some drivers have chosen but it's not really part of the generic HCI driver API (that the host or `hci_raw.c` see).